### DISCUSSION TITLE
Bolding/highlighting the search term

### DISCUSSION THREAD

**Discussion by silverwings15** [benbusby/whoogle-search#450] [Q&A]
Hi, I understand Google doesn't highlight the search term, but is it possible for Whoogle to do so?

**Comment by DUOLabs333** [comment]
It depends -- with Google's new algorithm, it sometimes doesn't search verbatim what you put in, so it might appear in a result's "preview" or not. Whoogle only displays what you would get from searching on Google proper. This may work for one-word terms, but how would it work with more than one term -- do you want all instances of every part of the word highlighted, or only the phrase?

**Reply by silverwings15** [reply]
i think old Google behavior (which Searx adopts) is to highlight all individual words that's being searched even if they're separated, which makes sense because you can append quote marks to the query if you want to search for the exact phrase.

**Comment by DUOLabs333** [comment]
I'm looking through the code, and it seems like right now, it can't handle formatted text (italicized, underlined, etc) -- it treats results (after cleaning it up of course) as plain text.

**Reply by silverwings15** [reply]
Thanks for the find! Unfortunate to hear though, hope it will be implemented in the future

**Reply by benbusby** [reply]
It does treat results as plaintext, but that doesn't have any affect on this being implemented. Essentially what you'd need to do is split the query into individual words as part of a regex, and replace all text that matches in the result body, but re-wrapped within a new `b` tag (or something). It would be similar to how the app currently replaces all instances of URLs when "site alts" are enabled in the config:

https://github.com/benbusby/whoogle-search/blob/19e89de5d914ded63e25d64118243fd9299b6ad5/app/filter.py#L329-L342

Example of how that would look:
<img width="746" alt="bold-demo" src="https://user-images.githubusercontent.com/33362396/138610481-c55cae2e-ee26-48ae-84ca-80a5f117f229.png">

Currently I'm focused on testing out a new refactor of how sessions are handled on the backend, but I'll add an issue to the backlog. Maybe someone will come along and implement it before I can get to it.

**Reply by benbusby** [reply]
(Unmarked the original answer since it doesn't completely encapsulate the full story, nothing personal!)

**Comment by DUOLabs333** [comment]
Maybe @benbusby can weigh in if I am mistaken?

**Reply by benbusby** [reply]
Thanks for pinging me!

**Comment by DUOLabs333** [comment]
I'm interested in making this. So is link_desc[0] just a string? What variable is the search term located in?

**Reply by benbusby** [reply]
`link_desc[0]` is a [NavigableString](https://www.crummy.com/software/BeautifulSoup/bs4/doc/#navigablestring) from BeautifulSoup, the library used for modifying the result tree. In this case you wouldn't want to immediately grab index 0, but rather iterate through the full list of NavigableStrings returned by `<soup element>.find_all` and then do a "replace all" of individual query terms with a new bold tag with the query inside. 

The search term is returned as part of `Search.new_search_query` in the `/search` endpoint. You'd likely want to pass that value (stored as just `query` in the search route) to whatever function ends up being created to modify the search results with the new bold tags.

**Reply by benbusby** [reply]
I'll go ahead and assign the new issue I created to you so that nobody else jumps on it. If you change your mind about working on it or anything, just let me know! Thank you!
